### PR TITLE
Gradle plugin: use `PatternSet` instead of `FileCollection` for inputs

### PR DESCRIPTION
**Pull request by petertrr**
### What's done:
* Changed logic
* Changed examples
* Changed tests
* Added proper annotations for some properties inside gradle plugin (resulting in 'automatic' `NO_SOURCE` statuses wen provided patterns don't match anything)
* Fix deprecation warnings in `diktat-gradle-plugin/build.gradle.kts`

This PR proposes a more convenient API for configuring inputs for gradle task. See #1026 , #1094

### CODE REVIEW

File: diktat-gradle-plugin/src/main/kotlin/org/cqfn/diktat/plugin/gradle/DiktatJavaExecTaskBase.kt [hunk 1]
```diff
@@ -42,6 +51,32 @@ open class DiktatJavaExecTaskBase @Inject constructor(
...
+     */
+    @get:IgnoreEmptyDirectories
+    @get:SkipWhenEmpty
+    @get:PathSensitive(PathSensitivity.RELATIVE)
+    @get:InputFiles
+    val actualInputs: FileCollection by lazy {
+        with(diktatExtension) {
+            // validate configuration
+            require(inputs == null && excludes == null) {
+                "`inputs` and `excludes` for diktat should be changed for inputs(Action)"
```
Comment (akuleshov7) [inline diktat-gradle-plugin/src/main/kotlin/org/cqfn/diktat/plugin/gradle/DiktatJavaExecTaskBase.kt:65]: "`inputs` and `excludes`  arguments for diktat task are deprecated and now should be changed for inputs(Action). Please check the Readme.md for more info"

File: .github/workflows/build_and_test.yml [hunk 2]
```diff
@@ -110,7 +117,7 @@ jobs:
     runs-on: ${{ matrix.os }}
     strategy:
       matrix:
-        os: [ ubuntu-latest, windows-latest, macos-latest ]
```
Comment (kgevorkyan) [inline .github/workflows/build_and_test.yml:113]: Why?

Comment (petertrr) [reply .github/workflows/build_and_test.yml:113]: We had a duplicated execution on ubuntu: main job (`build and test and upload codecov`) is basically the same, but also with codecov. So there is no need for a separate build_and_test on ubuntu

**Review by akuleshov7 — APPROVED** [review]

**Comment by akuleshov7** [comment]
Have you found out why tests did not catch this?

**Comment by petertrr** [comment]
> Have you found out why tests did not catch this?

The logic itself was actually correct - we were running ktlint with expected patterns. However, they must've broken something with pattern matching on windows. In our tests we were comparing command line (where patterns were correct), and in functional tests we are excluding files by specific names instead of wildcards only.

**Pull request merged**